### ISSUE TITLE
Out of domain for acos

### ISSUE THREAD

**Issue by stscieisenhamer** [spacetelescope/jwst#2746]
# Issue

From WFR2-OTB-20181004 processing run doggett/tests/wfr2/cstring/JWSTDP-2018_2-181017, running `calwebb_image3.cfg` gives the following:

<details><summary>Show log...</summary>
<p>

```
$ strun cfgs/calwebb_image3.cfg jw00759-o014_20181019t220201_image3_002_asn.json
2018-10-22 16:04:41,127 - stpipe.Image3Pipeline - INFO - Image3Pipeline instance created.
2018-10-22 16:04:41,129 - stpipe.Image3Pipeline.tweakreg - INFO - TweakRegStep instance created.
2018-10-22 16:04:41,130 - stpipe.Image3Pipeline.skymatch - INFO - SkyMatchStep instance created.
2018-10-22 16:04:41,132 - stpipe.Image3Pipeline.outlier_detection - INFO - OutlierDetectionStep instance created.
2018-10-22 16:04:41,133 - stpipe.Image3Pipeline.resample - INFO - ResampleStep instance created.
2018-10-22 16:04:41,134 - stpipe.Image3Pipeline.source_catalog - INFO - SourceCatalogStep instance created.
2018-10-22 16:04:41,134 - stpipe - INFO - Hostname: iljwdmscdp1.stsci.edu
2018-10-22 16:04:41,134 - stpipe - INFO - OS: Linux
2018-10-22 16:04:41,163 - stpipe.Image3Pipeline - INFO - Step Image3Pipeline running with args ('jw00759-o014_20181019t220201_image3_002_asn.json',).
2018-10-22 16:04:42,630 - stpipe.Image3Pipeline - INFO - Prefetching reference files for dataset: 'jw00759014001_02105_00001_nrcalong_cal.fits' reftypes = ['drizpars']
2018-10-22 16:04:43,057 - stpipe.Image3Pipeline - INFO - Prefetch for DRIZPARS reference file is '/grp/crds/cache/references/jwst/jwst_nircam_drizpars_0001.fits'.
2018-10-22 16:04:44,345 - stpipe.Image3Pipeline - INFO - Prefetching reference files for dataset: 'jw00759014002_02105_00001_nrcalong_cal.fits' reftypes = ['drizpars']
2018-10-22 16:04:44,349 - stpipe.Image3Pipeline - INFO - Prefetch for DRIZPARS reference file is '/grp/crds/cache/references/jwst/jwst_nircam_drizpars_0001.fits'.
2018-10-22 16:04:45,341 - stpipe.Image3Pipeline - INFO - Prefetching reference files for dataset: 'jw00759014001_02101_00001_nrcalong_cal.fits' reftypes = ['drizpars']
2018-10-22 16:04:45,345 - stpipe.Image3Pipeline - INFO - Prefetch for DRIZPARS reference file is '/grp/crds/cache/references/jwst/jwst_nircam_drizpars_0001.fits'.
2018-10-22 16:04:46,437 - stpipe.Image3Pipeline - INFO - Prefetching reference files for dataset: 'jw00759014001_02104_00001_nrcalong_cal.fits' reftypes = ['drizpars']
2018-10-22 16:04:46,440 - stpipe.Image3Pipeline - INFO - Prefetch for DRIZPARS reference file is '/grp/crds/cache/references/jwst/jwst_nircam_drizpars_0001.fits'.
2018-10-22 16:04:47,749 - stpipe.Image3Pipeline - INFO - Prefetching reference files for dataset: 'jw00759014002_02101_00001_nrcalong_cal.fits' reftypes = ['drizpars']
2018-10-22 16:04:47,753 - stpipe.Image3Pipeline - INFO - Prefetch for DRIZPARS reference file is '/grp/crds/cache/references/jwst/jwst_nircam_drizpars_0001.fits'.
2018-10-22 16:04:48,995 - stpipe.Image3Pipeline - INFO - Prefetching reference files for dataset: 'jw00759014002_02102_00001_nrcalong_cal.fits' reftypes = ['drizpars']
2018-10-22 16:04:48,999 - stpipe.Image3Pipeline - INFO - Prefetch for DRIZPARS reference file is '/grp/crds/cache/references/jwst/jwst_nircam_drizpars_0001.fits'.
2018-10-22 16:04:50,364 - stpipe.Image3Pipeline - INFO - Prefetching reference files for dataset: 'jw00759014002_02104_00001_nrcalong_cal.fits' reftypes = ['drizpars']
2018-10-22 16:04:50,369 - stpipe.Image3Pipeline - INFO - Prefetch for DRIZPARS reference file is '/grp/crds/cache/references/jwst/jwst_nircam_drizpars_0001.fits'.
2018-10-22 16:04:51,618 - stpipe.Image3Pipeline - INFO - Prefetching reference files for dataset: 'jw00759014001_02102_00001_nrcalong_cal.fits' reftypes = ['drizpars']
2018-10-22 16:04:51,622 - stpipe.Image3Pipeline - INFO - Prefetch for DRIZPARS reference file is '/grp/crds/cache/references/jwst/jwst_nircam_drizpars_0001.fits'.
2018-10-22 16:04:51,623 - stpipe.Image3Pipeline - INFO - Starting calwebb_image3 ...
2018-10-22 16:04:55,705 - stpipe.Image3Pipeline - INFO - Aligning input images...
2018-10-22 16:04:55,776 - stpipe.Image3Pipeline.tweakreg - INFO - Step tweakreg running with args (<ModelContainer>,).
2018-10-22 16:04:57,990 - stpipe.Image3Pipeline.tweakreg - INFO - Detected 3 sources in jw00759014001_02105_00001_nrcalong_cal.fits.
2018-10-22 16:05:00,072 - stpipe.Image3Pipeline.tweakreg - INFO - Detected 3 sources in jw00759014002_02105_00001_nrcalong_cal.fits.
2018-10-22 16:05:02,134 - stpipe.Image3Pipeline.tweakreg - INFO - Detected 3 sources in jw00759014001_02101_00001_nrcalong_cal.fits.
2018-10-22 16:05:04,144 - stpipe.Image3Pipeline.tweakreg - INFO - Detected 3 sources in jw00759014001_02104_00001_nrcalong_cal.fits.
2018-10-22 16:05:06,126 - stpipe.Image3Pipeline.tweakreg - INFO - Detected 3 sources in jw00759014002_02101_00001_nrcalong_cal.fits.
2018-10-22 16:05:08,145 - stpipe.Image3Pipeline.tweakreg - INFO - Detected 3 sources in jw00759014002_02102_00001_nrcalong_cal.fits.
2018-10-22 16:05:10,106 - stpipe.Image3Pipeline.tweakreg - INFO - Detected 3 sources in jw00759014002_02104_00001_nrcalong_cal.fits.
2018-10-22 16:05:12,061 - stpipe.Image3Pipeline.tweakreg - INFO - Detected 3 sources in jw00759014001_02102_00001_nrcalong_cal.fits.
----------------------------------------------------------------------
ERROR RUNNING STEP 'Image3Pipeline':
    Out of domain for acos
Traceback (most recent call last):
  File "/ifs/int/jwstc/info/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.13.0a0.dev67+gf56c3fd4-py3.6-linux-x86_64.egg/jwst/stpipe/cmdline.py", line 297, in step_from_cmdline
    step.run(*positional)
  File "/ifs/int/jwstc/info/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.13.0a0.dev67+gf56c3fd4-py3.6-linux-x86_64.egg/jwst/stpipe/step.py", line 400, in run
    step_result = self.process(*args)
  File "/ifs/int/jwstc/info/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.13.0a0.dev67+gf56c3fd4-py3.6-linux-x86_64.egg/jwst/pipeline/calwebb_image3.py", line 70, in process
    input_models = self.tweakreg(input_models)
  File "/ifs/int/jwstc/info/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.13.0a0.dev67+gf56c3fd4-py3.6-linux-x86_64.egg/jwst/stpipe/step.py", line 400, in run
    step_result = self.process(*args)
  File "/ifs/int/jwstc/info/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.13.0a0.dev67+gf56c3fd4-py3.6-linux-x86_64.egg/jwst/tweakreg/tweakreg_step.py", line 114, in process
    wcsimlist = list(map(self._imodel2wcsim, g))
  File "/ifs/int/jwstc/info/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.13.0a0.dev67+gf56c3fd4-py3.6-linux-x86_64.egg/jwst/tweakreg/tweakreg_step.py", line 170, in _imodel2wcsim
    meta={'image_model': image_model}
  File "/ifs/int/jwstc/info/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.13.0a0.dev67+gf56c3fd4-py3.6-linux-x86_64.egg/jwst/tweakreg/wcsimage.py", line 361, in __init__
    self.catalog = catalog
  File "/ifs/int/jwstc/info/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.13.0a0.dev67+gf56c3fd4-py3.6-linux-x86_64.egg/jwst/tweakreg/wcsimage.py", line 460, in catalog
    self.calc_bounding_polygon()
  File "/ifs/int/jwstc/info/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.13.0a0.dev67+gf56c3fd4-py3.6-linux-x86_64.egg/jwst/tweakreg/wcsimage.py", line 707, in calc_bounding_polygon
    self._calc_cat_convex_hull()
  File "/ifs/int/jwstc/info/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.13.0a0.dev67+gf56c3fd4-py3.6-linux-x86_64.egg/jwst/tweakreg/wcsimage.py", line 694, in _calc_cat_convex_hull
    self._poly_area = np.fabs(self._polygon.area())
  File "/ifs/int/jwstc/info/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/spherical_geometry/polygon.py", line 1096, in area
    area += subpoly.area()
  File "/ifs/int/jwstc/info/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/spherical_geometry/polygon.py", line 543, in area
    points[2:], degrees=False)
  File "/ifs/int/jwstc/info/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/spherical_geometry/great_circle_arc.py", line 320, in angle
    angle = math_util.angle(A, B, C)
ValueError: Out of domain for acos

----------------------------------------------------------------------
Traceback (most recent call last):
  File "/ifs/int/jwstc/info/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.13.0a0.dev67+gf56c3fd4-py3.6-linux-x86_64.egg/EGG-INFO/scripts/strun", line 26, in <module>
    step = Step.from_cmdline(sys.argv[1:])
  File "/ifs/int/jwstc/info/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.13.0a0.dev67+gf56c3fd4-py3.6-linux-x86_64.egg/jwst/stpipe/step.py", line 162, in from_cmdline
    return cmdline.step_from_cmdline(args)
  File "/ifs/int/jwstc/info/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.13.0a0.dev67+gf56c3fd4-py3.6-linux-x86_64.egg/jwst/stpipe/cmdline.py", line 297, in step_from_cmdline
    step.run(*positional)
  File "/ifs/int/jwstc/info/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.13.0a0.dev67+gf56c3fd4-py3.6-linux-x86_64.egg/jwst/stpipe/step.py", line 400, in run
    step_result = self.process(*args)
  File "/ifs/int/jwstc/info/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.13.0a0.dev67+gf56c3fd4-py3.6-linux-x86_64.egg/jwst/pipeline/calwebb_image3.py", line 70, in process
    input_models = self.tweakreg(input_models)
  File "/ifs/int/jwstc/info/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.13.0a0.dev67+gf56c3fd4-py3.6-linux-x86_64.egg/jwst/stpipe/step.py", line 400, in run
    step_result = self.process(*args)
  File "/ifs/int/jwstc/info/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.13.0a0.dev67+gf56c3fd4-py3.6-linux-x86_64.egg/jwst/tweakreg/tweakreg_step.py", line 114, in process
    wcsimlist = list(map(self._imodel2wcsim, g))
  File "/ifs/int/jwstc/info/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.13.0a0.dev67+gf56c3fd4-py3.6-linux-x86_64.egg/jwst/tweakreg/tweakreg_step.py", line 170, in _imodel2wcsim
    meta={'image_model': image_model}
  File "/ifs/int/jwstc/info/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.13.0a0.dev67+gf56c3fd4-py3.6-linux-x86_64.egg/jwst/tweakreg/wcsimage.py", line 361, in __init__
    self.catalog = catalog
  File "/ifs/int/jwstc/info/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.13.0a0.dev67+gf56c3fd4-py3.6-linux-x86_64.egg/jwst/tweakreg/wcsimage.py", line 460, in catalog
    self.calc_bounding_polygon()
  File "/ifs/int/jwstc/info/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.13.0a0.dev67+gf56c3fd4-py3.6-linux-x86_64.egg/jwst/tweakreg/wcsimage.py", line 707, in calc_bounding_polygon
    self._calc_cat_convex_hull()
  File "/ifs/int/jwstc/info/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.13.0a0.dev67+gf56c3fd4-py3.6-linux-x86_64.egg/jwst/tweakreg/wcsimage.py", line 694, in _calc_cat_convex_hull
    self._poly_area = np.fabs(self._polygon.area())
  File "/ifs/int/jwstc/info/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/spherical_geometry/polygon.py", line 1096, in area
    area += subpoly.area()
  File "/ifs/int/jwstc/info/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/spherical_geometry/polygon.py", line 543, in area
    points[2:], degrees=False)
  File "/ifs/int/jwstc/info/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/spherical_geometry/great_circle_arc.py", line 320, in angle
    angle = math_util.angle(A, B, C)
ValueError: Out of domain for acos
```

</p>
</details>

**Comment by stscieisenhamer** [comment]
# Resources

Data is in `/grp/jwst/ssb/eisenhamer/jwst_issues/issue2746`.

**Comment by mcara** [comment]
@jdavies-st I'll add a graceful handling of the case with no sources

**Comment by jdavies-st** [comment]
@larrybradley, do any of these "sources" look to be 10 sigma?  I suspect that the source detection should not be detecting anything here.

@hbushouse, SDP does use the updated .cfg files to run the new version of the pipeline, right?

**Comment by hbushouse** [comment]
@jdavies-st How recently was the change made from 2.5 to 10 sigma? Was it included in our 0.12.0 delivery on 10/16/2018?

**Comment by jdavies-st** [comment]
Yes, it is in 0.12.0.  It was made over a month ago in #2510.  During the image registration hack day, we found not only inconsistency in the `spec` vs defaults in the `tweakreg.cfg` file, we found the low sigma a big problem for both NIRCam and MIRI imaging data.

**Comment by hbushouse** [comment]
OK, then it should be in use in DMS. Even if they aren't using the cfg files, the default value in the tweakreg_step has also been set to 10, so either way it should be using 10.

**Comment by hbushouse** [comment]
How is the snr computed in tweakreg (or whatever photutils routines are being called)? Is it using values from the input ERR arrays? If so, that could be the problem, because those values are likely to not be at all realistic, especially for test data.

**Comment by jdavies-st** [comment]
Yes, so the issue here is that there is somewhere in the image that `photutils` source detection thinks is a 10-sigma detection.  And given the images are just striped, blank fields, then the probably points to an unresolved issue in source detection.  I suspect.

**Comment by larrybradley** [comment]
I'll take a look.

**Comment by stscieisenhamer** [comment]
FYI: For all issues, such as this, where I provide a log, that is usually running it myself using condadev. So the log represents state-of-art, unless otherwise indicated. This is done in case something has already been fixed, different environment, etc., for the exact reasons discussed already.

**Comment by hbushouse** [comment]
Fixed by #2796 and #2797
